#### Que1. Write a program to print “Hello Python” ten times.

In [5]:
i=1  # --> initialization

while i<=10: #--> Condition
    print(i,'Hello Python') #--> Print Statement
    i+=1 # --> Steps of Loops

1 Hello Python
2 Hello Python
3 Hello Python
4 Hello Python
5 Hello Python
6 Hello Python
7 Hello Python
8 Hello Python
9 Hello Python
10 Hello Python


In [6]:
for i in range(1,11): # Range with start and Stop Position.
    print(i,"Hello Python") # Print Statement

1 Hello Python
2 Hello Python
3 Hello Python
4 Hello Python
5 Hello Python
6 Hello Python
7 Hello Python
8 Hello Python
9 Hello Python
10 Hello Python


Explanation: In this Program, The Hello is printed with the use of while and for loop. In while loop, the steps are mentioned in the code itself. we have taken the condition as "i" should be less than or equals to 10 as we need to print "Hello" 10 times. 
In For Loop, the range is taken as 1(start position) to 11(stop position). The stop position is 11 as we need to print the "Hello" 10 times and in range the last position is not included.


#### Que2.  Write a program to print number from 1 to 10.

In [19]:
i=1  # --> initialization
while i<=10: #--> Condition
    print(i,end = " , ") #--> Print Statement
    i+=1  # --> Steps of Loops

1 , 2 , 3 , 4 , 5 , 6 , 7 , 8 , 9 , 10 , 

In [20]:
for i in range(1,11): # Range with start and Stop Position.
    print(i,end = " , ") # Print Statement

1 , 2 , 3 , 4 , 5 , 6 , 7 , 8 , 9 , 10 , 

Explanation: In this Program, The Hello is printed with the use of while and for loop. In while loop, the steps are mentioned in the code itself. we have taken the condition as "i" should be less than or equals to 10 as we need to print number from 1 to 10. 
In For Loop, the range is taken as 1(start position) to 11(stop position). The stop position is 11 as we need to print the numbers from 1 to 10 and in range the last position is not included.


#### Que3.  Write a program to print sum of the first ten numbers.

In [10]:
add=0  
i=1
while i<=10:
    add=add+i
    i=i+1

print("The sum of the first ten numbers is",add)

The sum of the first ten numbers is 55


Explanation: In this Program, we need to print the sum of first 10 numbers. So, initially zero is stored in the add. The i starts from 1 and ends at 10. Arithmetic operator (Addition) is used. Inside the loop, it adds each integer to the add variable. Finally, it prints out the addition of the first ten numbers.

#### Que4.  Write a program to print n number entered by the user.

In [18]:
n = int(input("Enter a number.:"))

for i in range(1, n+1):
    print(i,end=" , ")

Enter a number.:5
1 , 2 , 3 , 4 , 5 , 

Explanation: In this program, user input is asked. Then, for loop to iterate over the range from 1 to n+1, printing each number on a separate line using the print() function.

#### Que5.  Write a program to print sum of n numbers.

In [9]:
n = int(input("Enter n:"))
add=0
i=1
while i<=n:
    add=add+i
    i=i+1
print("The sum of the", n, "integers is:",add)

Enter n:10
The sum of the 10 integers is: 55


Explanation: In this Program, Firstly, User input is asked. Then, initially zero is stored in the add. The i starts from 1 and ends at n. Arithmetic operator (Addition) is used. Inside the loop, it adds each integer to the add variable. Finally, it prints out the addition of the n numbers.

#### Que6.  Write a program to print table of the numbers.

In [10]:
num = int(input("Enter a number: "))

for i in range(1, 11):
    print(num, "x", i, "=", num*i)

Enter a number: 5
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


Explanation: In this Program, Firstly, User input is asked. In for loop, the range is set as 1(Start Position) and 11(Last Position). In print statement, Arithmetic operator "*" is used to get multiplication of the given number.

#### Que7.  Write a program to calculate the factorial of the numbers.

In [21]:
n = int(input("enter the n for which you want factorial.:"))
fac = 1
while n >= 1:
    fac = fac * n
    n = n - 1
print(fac)

enter the n for which you want factorial.:5
120


Explanation: Factorial of a positive integer (number) is the sum of multiplication of all the integers smaller than that positive integer. For example, factorial of 5 is 5 * 4 * 3 * 2 * 1 which equals to 120. 
In this Program, a user input is asked. Then, While loop starts from 1 to n, multiplying each number with the factorial variable. Finally, it prints out the factorial of n.


#### Que8.  Write a program to print even number between 1 to 100.

In [21]:
i = 2
while i<=100:
    print(i,end=" ")
    i+=2

2 4 6 8 10 12 14 16 18 20 22 24 26 28 30 32 34 36 38 40 42 44 46 48 50 52 54 56 58 60 62 64 66 68 70 72 74 76 78 80 82 84 86 88 90 92 94 96 98 100 

Explanation: In this program, initializes is set as i to 2, and then while loop is used to repeatedly print the value of i and increment it by 2 until it reaches 100.And print Function is used to print value of i after each iteration.

#### Que9.  Write a program to print odd number between 1 to 100.

In [22]:
i = 1
while i<=100:
    print(i,end=" ")
    i+=2

1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 51 53 55 57 59 61 63 65 67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99 

Explanation: In this program, initializes is set as i to 1, and then while loop is used to repeatedly print the value of i and increment it by 2 until it reaches 100.And print Function is used to print value of i after each iteration.

#### Que10.  Write a program to check whether a given number is palindrome or not.

In [24]:
i = int(input("Enter a number.:"))
rev = 0
x = i # x stores the value of i as after the reverse is happened, the i value will reduced to 0.

while i > 0:
    rev = (rev*10) + (i%10)   # rev = (0*10) + (525%10) == (0 + 5) ==  5
    i= i//10  # i = 525/10 = 52.2 and using // it ignores the decimal part that's why i = 52
if x == rev:
    print("The given number is Palindrome.")
else:
    print("The given number is not Palindrome.")

Enter a number.:525
The given number is Palindrome.


Explanation: A palindrome is a word, number, phrase, or other sequence of symbols that reads the same backwards as forwards, such as madam or racecar, the date and time 12/21/33 12:21, and the sentence: "A man, a plan, a canal – Panama". To know whether the given number is Palindrome or not, first, we need to reverse the number. and after that if the reversed value is equals to the original value than the given number is Palindrome. Here, we have stored "i" into "x" as in reverse logic the i value becomes "0" in the end so after the reversed operation is done i becomes "0" and then there will be no original value stored in n as it becomes 0. So to compare the reverse value with original value we have used x which stores the original value.The // operator is used for integer division, and the % operator is used to get the remainder when the number is divided by 10.

#### Que11.  Write a program to find factors of the numbers.

In [28]:
## Factors of a number.

num = int(input("Enter a number.:"))

for i in range(1, num+1):
    if num % i == 0:
        print("The factors of given number is ",i)

Enter a number.:12
The factors of given number is  1
The factors of given number is  2
The factors of given number is  3
The factors of given number is  4
The factors of given number is  6
The factors of given number is  12


Explanation: A factor is a number that divides another number, leaving no remainder. The range starts with 1 and stop is increment by 1 as we are taking user input. the "if" conditions say, if the entered number is divisible by i and gives "0" as a remainder, the entered number will have factors and it is printed in print statement. In this program, Firstly User input is asked. Then, for loop is used Which starts from 1 to n, checking whether each number is a factor of n by using the modulus operator %. If the remainder is zero, then the number is a factor of n and then it is printed.

#### Que12.  Write a program to print the reverse of the number.

In [21]:
## Reverse of the number

i = int(input("Enter a number.:"))
rev = 0

while i > 0:
    rev = (rev*10) + (i%10)   # rev = (0*10) + (287%10) == (0 + 7) ==  7
    i= i//10                  # i = 287/10 = 28.7 and using // it ignores the decimal part that's why i = 28
print("Reverse of a given number is",rev)

Enter a number.:287
Reverse of a given number is 782


Explanation: In this program, "i" stores the user input, reverse is stored as "0" initially. The condition says i should be greater than 0. Reverse formula is explained in code itself. Finally, print function is used to print the result.

#### Que13.  Write a program to print the sum of a reverse of a number.

In [15]:
i = int(input("Enter a number.:"))

## Calculate the Reverse of the number
rev = 0
x = i
while i > 0:
    rev = (rev*10) + (i%10)   # rev = (0*10) + (287%10) == (0 + 7) ==  7
    i= i//10                  # i = 287/10 = 28.7 and using // it ignores the decimal part that's why i = 28
print("Reverse of a given number is",rev)

## Calculate the sum of the number and the reverse of the number,
sum = x + rev

print("The sum of a reverse of a number is",sum)


Enter a number.:123
Reverse of a given number is 321
The sum of a reverse of a number is 444


Explanation: In this program, First we need to calculate the reverse of the number and then we need to add the number and the reverse of the number. So "i" stores the user input, reverse is stored as "0" initially. The condition says i should be greater than 0. Reverse formula is explained in code itself. Finally, print function is used to print the result. After calculating the reverse, the program calculates the sum of the original number and its reverse, and stores the result in a variable called sum. Finally, it prints the sum of the number and its reverse.

#### Que14.  Write a program to check whether a given number is prime or not.

In [4]:
n = int(input("Enter a Number.:"))
count = 0
i = 1
while i<=n:
    if n%i==0:
        count+=1
    i+=1    
if count == 2:
    print(" The Given number is Prime.")
else:
    print(" The Given number is not Prime.")

Enter a Number.:13
 The Given number is Prime.


Explanation: A whole number greater than 1 that cannot be exactly divided by any whole number other than itself and 1 (e.g. 2, 3, 5, 7, 11). Prime Numbers always have only 2 Factors. 

In this program, user input is asked. "Count" is set as zero and "i" is set as 1. while loop is used, it run from the 1 to the given number. And also checks whether the number of factorials is equal to 2 or not. if else condition is also used. if the count is equals to 2, the given number is prime and if not then the given number is not prime.

#### Que15.  Write a program to print prime number between 1 to 100.

In [1]:
n = 1
while n<=100:
    i = 2
    while i<=n//2:
        if n%i ==0:
            break
        i+=1
    if i>n//2:
        print(n,end = " ")
    n+=1

1 2 3 5 7 11 13 17 19 23 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 

In this program, “n” starts with 1 and ends at 100. "i" is set as 2.After that the condition is checked, where "i" should be less than or equals to "n//2". Inside the loop, it is checking for whether the given input is fulling divisible by i or not. Now we have broken the loop, and it will start away go to the if condition which says i should be greater than n//2 then it will print the numbers else it will not print the numbers.

#### Que16.  Write a program to print number in reverse order entered by the user.

In [16]:
## Reverse of the number

i = int(input("Enter a number.:"))
rev = 0

while i > 0:
    rev = (rev*10) + (i%10)   # rev = (0*10) + (287%10) == (0 + 7) ==  7
    i= i//10                  # i = 287/10 = 28.7 and using // it ignores the decimal part that's why i = 28
print("Reverse of a given number is",rev)

Enter a number.:123
Reverse of a given number is 321


Explanation: In this program, "i" stores the user input, reverse is stored as "0" initially. The condition says i should be greater than 0. Reverse formula is explained in code itself. Finally, print function is used to print the result.

#### Que17.  Write a program to print the cube of all the numbers from 1 to the given number.

In [1]:
n = int(input("Enter a number: "))

# iterate from 1 to n and print the cube of each number
print("Cube of numbers from 1 to", n)
for num in range(1, n+1):
    cube = num**3
    print(num, "cube is", cube)

Enter a number: 3
Cube of numbers from 1 to 3
1 cube is 1
2 cube is 8
3 cube is 27


In this program, the user first enters a number, which is stored in the variable n. We then use a for loop to iterate from 1 to n, and print the cube of each number. Inside the loop, we calculate the cube of the current number using the exponent operator ** and store it in the variable cube. We then use the print() function to output the number and its cube. Finally, the program uses the print() function to output the list of cubes of all the numbers from 1 to the given number.

#### Que18.  Write a program to count the total numbers of digits in a number.

In [11]:
n = int(input("Enter A Number.:"))
count = 0
while n>0:
    n=n//10 ## To get updated value in each iteration.
    ## working: 12345 ==> 1234 ==>123 ==> 12==> 1 ==> 0 ,Runs until the value becomes 0.
    count = count + 1
print("Total numbers of digits in this number is ",count)

Enter A Number.:12345
Total numbers of digits in this number is  5


In this program, firstly, a user input is asked and then count is set as zero initially. Then while loop is used to count the number of digits in the number, by repeatedly dividing the number by 10 and incrementing the count by 1 for each digit. The loop continues until the number becomes zero. Print statement is used to print the count.

#### Que19.  Write a program to print Fibonacci series.

In [8]:
n = int(input("Enter a Number.:"))
x=0
y=1
z=0
while z<=n: # z = 0 as we have set the value 0 in z, so condition is true.
    print(z)  #print z as 0
    x=y  # x=y i.e x=1 as y is set as 1.
    y=z  # y = z i.e y = 0.
    z=x+y
## z = 1 + 0 = 1. as it is loop, it will go the condition statement and this will be done until the conditions become false.


Enter a Number.:10
0
1
1
2
3
5
8


Explanation: The Fibonacci series is a sequence of numbers in which each number after the first two is the sum of the two preceding ones. The first two numbers in the series are typically defined as either 0 and 1 or 1 and 1, depending on the context. The sequence goes like this: 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, and so on. In this program, Swap is used. The whole mechanism is explained the code itself.

#### Que20.  Write a program to check whether a given number is Armstrong number or not.

In [3]:
n = int(input("Enter a Number.:"))
i = n
count = 0
while i > 0:  ## This will count the value of Number.
        count = count + 1
        i = i//10
sum = 0        
i = n
while i > 0:  ## This will set the digit.
    digit = i % 10
    x = 1
    product = 1
    while x <= count:  ## This loop is used to get the product.
        product = product * digit
        x = x + 1
    sum = sum + product
    i = i // 10
if sum == n:
    print("A given number is Armstrong number.")
else:
    print("A given number is Not Armstrong number.")
        
            

Enter a Number.:153
A given number is Armstrong number.


Explanation: An Armstrong number is a number that is equal to the sum of its own digits raised to the power of the number of digits. For example, 153 is an Armstrong number because it has three digits and it can be written as 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153.

In this Program, Firstly, user input is asked which is stored in "n". Then we need to set a variable as I, we store the n value, as "n" is going to become zero. Then we need to first find the count of number of digits as to know the product number. Then we need to calculate the digit and then multiplication is also done. Product is set as 1 as there is no difference in multiplying the number with 1. Then Addition is done. Sum is set as zero initially. Then if else condition is used. Condition says if the sum is equals to original value then the given number is Armstrong number and if not, it is not a Armstrong number.


#### Que21.  Write a program to check whether a given number is strong  number or not.

In [20]:
num = input("Enter a Number.:")
add = 0

for i in num:  # This loop will separate the numbers as 1,4,5.
    i = int(i) # convert the string into integer.
    fact = 1
    for j in range(1,i+1): # This loop will calculate the factorial and Add the Numbers.
        fact = fact * j 
    add = add + fact
if add == int(num):
    print("A given number is Strong number. ")
else:
    print("A given number is not Strong number. ")

Enter a Number.:145
A given number is Strong number. 


Explanation: A strong number is a number that is equal to the sum of the factorial of its digits. For example, 145 is a strong number because 1! + 4! + 5! = 1 + 24 + 120 = 145.

In this Program, firstly user input is asked in string and stored in num. Add is set as zero initially. Firstly, we need to separate the number from each other, that’s why the input is taken into string. Later for calculations we have converted the string to integer. Then second loop is for calculating the factorial of number and then the addition of factorial is also done in same loop. If else condition is used. If the added value is equals to the original value, then the given number is strong number and else it is not a strong number. Here, in condition the “num” is converted into integer as the string value is always not equals to the integer.


#### Que22.  Write a program to check whether a given number is perfect number or not.

In [4]:
num = int(input("Enter a Number.:"))
res = 0

for i in range(1,num):
    if num%i == 0: 
        res=res+i
if res == num:
    print(num, "is a Perfect Number.")
else:
    print(num,"is not a Perfect Number.")
    

Enter a Number.:6
6 is a Perfect Number.


Explanation:A perfect number is a positive integer that is equal to the sum of its proper divisors (excluding the number itself). For example, the number 6 is a perfect number because its proper divisors (excluding itself) are 1, 2, and 3, and their sum is equal to 6. Similarly, the number 28 is a perfect number because its proper divisors are 1, 2, 4, 7, and 14, and their sum is equal to 28.

In this program, firstly user input is asked and stored in num. Result is set as zero initially. Then loops starts from 1 to num//2, checking if i is a divisor of num using the modulo operator. If i is a divisor of num, it adds i to the sum. Then if else condition is used. Condition says if result is equals to original number, then the given number is  perfect number else it is not a  perfect number.

#### Que 23. Write a program to find those numbers which are divisible by 7 and multiple of 5, between 1000 to 2000.

In [2]:
# iterate from 1000 to 2000 and print the numbers that satisfy the condition
print("Numbers divisible by 7 and multiple of 5 between 1000 and 2000:")
for num in range(1000, 2001):
    if num % 7 == 0 and num % 5 == 0:
        print(num)

Numbers divisible by 7 and multiple of 5 between 1000 and 2000:
1015
1050
1085
1120
1155
1190
1225
1260
1295
1330
1365
1400
1435
1470
1505
1540
1575
1610
1645
1680
1715
1750
1785
1820
1855
1890
1925
1960
1995


In this program, for loop is used, which starts from 1000 to 2000 and print the numbers that are divisible by 7 and multiple of 5. Inside the loop, we use an if statement to check if the current number satisfies both conditions. We use the modulus operator % to check if the number is divisible by 7 and 5. Print is used to print the numbers from 1000 to 2000 which are divisible by 7 and 5.

#### Que 24. Write a program to print the following Number patterns using a loop.

In [80]:
## a. Number Pattern 1

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(n): # 'i' indicates the Rows.
    for j in range(i+1): # 'j' indicates the Columns.
        print(j+1,end = " ")
    print() # For Next Line

Enter the Numbers of rows.:5
1 
1 2 
1 2 3 
1 2 3 4 
1 2 3 4 5 


In [81]:
## b. Number Pattern 2

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(n): # 'i' indicates the Rows.
    for j in range(i,-1,-1): # 'j' indicates the Columns.
        print(i+1,end = " ")
    print() # For Next Line

Enter the Numbers of rows.:5
1 
2 2 
3 3 3 
4 4 4 4 
5 5 5 5 5 


In [16]:
## c. Number Pattern 3

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(n): # 'i' indicates the Rows.
    for j in range(n-i): # 'j' indicates the Columns.
        print(n-j,end = " ")
    print() # For Next Line

Enter the Numbers of rows.:5
5 4 3 2 1 
5 4 3 2 
5 4 3 
5 4 
5 


In [25]:
## d. Number Pattern 4

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.
for i in range(n, 0, -1):
    for j in range(1, i+1):
        print(j, end=" ")
    print()

Enter the Numbers of rows.:5
1 2 3 4 5 
1 2 3 4 
1 2 3 
1 2 
1 


In [23]:
## e. Number Pattern 5

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.
for i in range(n,0,-1):
    for j in range(i,0,-1):
        print(j, end=" ")
    print()

Enter the Numbers of rows.:5
5 4 3 2 1 
4 3 2 1 
3 2 1 
2 1 
1 


#### Que 25. Write a program to print the following star patterns using a loop.

In [68]:
## a. Right Half-Pyramid Pattern Program

num = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(num):
    for j in range(num-1-1):
        print(" ",end=" ")
    for j in range(i+1):
        print("*",end=" ")
    print()
        

Enter the Numbers of rows.:5
      * 
      * * 
      * * * 
      * * * * 
      * * * * * 


In [70]:
## b. Rectangle Shape

r = int(input("Enter the Numbers of Rows.:"))
c = int(input("Enter the Number of Columns.:"))

for i in range(r):
    for j in range(c):
        print("*",end = " ")
    print()


Enter the Numbers of Rows.:6
Enter the Number of Columns.:5
* * * * * 
* * * * * 
* * * * * 
* * * * * 
* * * * * 
* * * * * 


In [55]:
## c. Downward Half-Pyramid Pattern Program
num = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(num,0,-1): # 'i' indicates the Rows. 
    for j in range(0,i): # 'j' indicates the Columns.
        print("*",end = " ") # This will print * and end is used for not adding a new line.
    print()

Enter the Numbers of rows.:5
* * * * * 
* * * * 
* * * 
* * 
* 


In [63]:
## d. Downward Half-Pyramid Pattern Program

num = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(1,num+1): # 'i' indicates the Rows. 
    for j in range(1,num+1): # 'j' indicates the Columns.
        if j >= i:
            print("*",end = " ") # To print "*"
        else:
            print(" ",end = " ") # To Give a space.
    print()

Enter the Numbers of rows.:5
* * * * * 
  * * * * 
    * * * 
      * * 
        * 


In [78]:
## e. Left Half-Pyramid Pattern Program

num = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.

for i in range(1,num+1):
    for j in range(1,num+1):
        if j<=n-i:
            print(" ", end=" ")
        else:
            print("*", end=" ")
    print()

Enter the Numbers of rows.:6
          * 
        * * 
      * * * 
    * * * * 
  * * * * * 
* * * * * * 


#### Que 26. Write a program to print the following alphabet patterns using a loop.

In [101]:
## a. Alphabet pattern 1

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.
for i in range(1, n+1):
    for j in range(i):
        print(chr(65+j), end=" ")
    print()

Enter the Numbers of rows.:5
A 
A B 
A B C 
A B C D 
A B C D E 


In [109]:
## b. Alphabet pattern 2

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.
for i in range(0, n+1):
    for j in range(i+1):
        print(chr(65+i), end=" ")
    print()

Enter the Numbers of rows.:4
A 
B B 
C C C 
D D D D 
E E E E E 


In [125]:
## c. Alphabet pattern 3

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.
for i in range(n, 0, -1):
    for j in range(i):
        print(chr(65+j), end=" ")
    print()

Enter the Numbers of rows.:5
A B C D E 
A B C D 
A B C 
A B 
A 


In [134]:
## d. Alphabet pattern 4

n = int(input("Enter the Numbers of rows.:")) # Ask user input for rows.
for i in range(n, 0, -1):
    for j in range(i):
        print(chr(64+i), end=" ")
    print()

Enter the Numbers of rows.:5
E E E E E 
D D D D 
C C C 
B B 
A 
